### PR TITLE
Adding CIDER and clj-refactor support to project.clj

### PR DESCRIPTION
**Pull request by tdaudelin**
Guide followed: https://github.com/bhauman/lein-figwheel/wiki/Using-the-Figwheel-REPL-within-NRepl

### CODE REVIEW

**Comment by daudelin** [comment]
braid cider repls

**Comment by rafd** [comment]
Thanks @daudelin  This will surely help other contributors who use emacs.

Could you try extracting it out into its own profile, like this:

```clojure
:cider {:dependencies [[com.cemerick/piggieback "0.2.1"]]
        :repl-options {:nrepl-middleware [cemerick.piggieback/wrap-cljs-repl]}
        :plugins [[cider/cider-nrepl "0.15.0-SNAPSHOT"]
                  [refactor-nrepl "2.3.0-SNAPSHOT"]]}}
```

Then to run the project, you'll need to do: `lein with-profiles +cider repl`

**Comment by jamesnvc** [comment]
👍

**Comment by daudelin** [comment]
Good suggestion. Moved it to a profile and also added some documentation in the getting up and running page for how to use it.

**Comment by jamesnvc** [comment]
Two little things; you can make the cider profile inherit from the dev profile by writing it like `:cider [:dev {:dependencies ...}]`, which you probably want (not sure if it's strictly necessary, but I like to be explicit).

The other little style thing is for the markdown. I like to write markdown with each sentence on its own line, since it renders the same but makes for nicer diffs.

Thanks a lot for adding this!

File: docs/dev/how-to/getting-up-and-running-in-development.md [hunk 1]
```diff
@@ -157,6 +157,46 @@ By default, the Clojurescript REPL that starts with figwheel doesn't support com
   ```
 
 
+### The Best REPL
```
Comment (jamesnvc) [inline docs/dev/how-to/getting-up-and-running-in-development.md]: Maaaaybe say `Emacs Alternative` or something? 😛

Comment (daudelin) [inline docs/dev/how-to/getting-up-and-running-in-development.md]: Heh, couldn't resist given the section above 😆

**Pull request merged**